src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationMiddleware.cs

```diff
@@ -42,6 +43,7 @@ public class OpenIdConnectAuthenticationMiddleware : AuthenticationMiddleware<Op
             [NotNull] IDataProtectionProvider dataProtectionProvider,
             [NotNull] ILoggerFactory loggerFactory,
             [NotNull] IUrlEncoder encoder,
+            [NotNull] IDistributedCache cache,
```

Comment:
The trouble with adding IDistributedCache as a constructor parameter is that it makes it required. What if you took in an IServiceProvider instead and then tried to do an optional lookup of IDistributedCache?